server/src/main/java/org/opensearch/action/admin/indices/mapping/get/GetMappingsResponse.java

```diff
@@ -42,119 +43,82 @@
 import org.opensearch.common.io.stream.StreamOutput;
 import org.opensearch.common.xcontent.ToXContentFragment;
 import org.opensearch.common.xcontent.XContentBuilder;
-import org.opensearch.common.xcontent.XContentParser;
-import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.index.mapper.MapperService;
 
 import java.io.IOException;
-import java.util.Map;
-
-import static org.opensearch.rest.BaseRestHandler.DEFAULT_INCLUDE_TYPE_NAME_POLICY;
 
 public class GetMappingsResponse extends ActionResponse implements ToXContentFragment {
 
     private static final ParseField MAPPINGS = new ParseField("mappings");
 
-    private ImmutableOpenMap<String, ImmutableOpenMap<String, MappingMetadata>> mappings = ImmutableOpenMap.of();
+    private final ImmutableOpenMap<String, MappingMetadata> mappings;
 
-    public GetMappingsResponse(ImmutableOpenMap<String, ImmutableOpenMap<String, MappingMetadata>> mappings) {
+    public GetMappingsResponse(ImmutableOpenMap<String, MappingMetadata> mappings) {
         this.mappings = mappings;
     }
 
     GetMappingsResponse(StreamInput in) throws IOException {
         super(in);
         int size = in.readVInt();
-        ImmutableOpenMap.Builder<String, ImmutableOpenMap<String, MappingMetadata>> indexMapBuilder = ImmutableOpenMap.builder();
+        ImmutableOpenMap.Builder<String, MappingMetadata> indexMapBuilder = ImmutableOpenMap.builder();
         for (int i = 0; i < size; i++) {
-            String key = in.readString();
-            int valueSize = in.readVInt();
-            ImmutableOpenMap.Builder<String, MappingMetadata> typeMapBuilder = ImmutableOpenMap.builder();
-            for (int j = 0; j < valueSize; j++) {
-                typeMapBuilder.put(in.readString(), new MappingMetadata(in));
+            String index = in.readString();
+            if (in.getVersion().before(Version.V_2_0_0)) {
+                int mappingCount = in.readVInt();
+                assert mappingCount == 1 || mappingCount == 0 : "Expected 0 or 1 mappings but got " + mappingCount;
+                if (mappingCount == 1) {
+                    String type = in.readString();
+                    assert MapperService.SINGLE_MAPPING_NAME.equals(type) : "Expected type [_doc] but got [" + type + "]";
+                    indexMapBuilder.put(index, new MappingMetadata(in));
+                } else {
+                    indexMapBuilder.put(index, MappingMetadata.EMPTY_MAPPINGS);
```

Comment:
Should we add `mappingCount == 0` and throw `IllegalStateException` to be consistent with previous change?
```
              } else {
                    throw new IllegalStateException("Expected 0 or 1 mappings but got: " + numMappings);
                }
```

Reply:
agree.. I like the consistency. Done.